### PR TITLE
ponyc: only pour bottle when user has Xcode 8

### LINKED ISSUE #5346: ponyc 4.0 bottle on 10.11.6 appears to only work with Xcode 8
See #5343, specifically [this comment](https://github.com/Homebrew/homebrew-core/pull/5343#issuecomment-250327795).

If someone could test `brew install -s ponyc` on El Capitan with Xcode 7.3.1 that would be great (I don't have a VM for that).

### PR DESCRIPTION
**Pull request by zmwangx**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

See https://github.com/ponylang/ponyc/issues/1274 and https://github.com/Homebrew/homebrew-core/issues/5346.

TL;DR: `ponyc` 4.0 bottle doesn't work on systems with Xcode 7.3, but building from source on such systems is fine.

Fixes #5346.

### CODE REVIEW

File: Formula/ponyc.rb [hunk 1]
```diff
@@ -17,6 +17,15 @@ class Ponyc < Formula
   depends_on "pcre2"
   needs :cxx11
 
+  # https://github.com/ponylang/ponyc/issues/1274
+  # https://github.com/Homebrew/homebrew-core/issues/5346
+  pour_bottle? do
+    reason <<-EOS.undent
+    The bottle requires Xcode 8 to work properly.
```
Comment (MikeMcQuaid) [inline Formula/ponyc.rb]: Indent by two-characters. Also, technically isn't this checking for either the Xcode or CLT clang version so the messaging should be adjusted accordingly.

Comment (zmwangx) [inline Formula/ponyc.rb]: Actually I'm not sure what would happen on a CLT-only system. I'll check.

**Review by DomT4 — APPROVED** [review]

**Comment by SeanTAllen** [comment]
Thanks @zmwangx!
